modules/internal/ChapelLocale.chpl

```diff
@@ -182,34 +182,82 @@ module ChapelLocale {
   // Locale methods we want to have show up in chpldoc start here:
 
   /*
-    Get the hostname of this locale.
+    Get the hostname of the hardware associated with this locale.
+
+    The behavior of this method does not differ for calls made from
+    sub-locales (Unlike :proc:`locale.name`).
 
     :returns: the hostname of the compute node associated with the locale
     :rtype: string
+
   */
-  inline proc locale.hostname {
+  inline proc locale.hostname: string {
     return this._value.hostname;
   }
 
   /*
-    Get the name of this locale.  In practice, this is often the
-    same as the hostname, though in some cases (like when using
-    local launchers), it may be modified.
+    Get the name of this locale.
 
-    :returns: locale name
-    :rtype: string
+    In general, this method returns the same string as :proc:`locale.hostname`;
+    however, it can differ when called on a sub-locale (such as a GPU),
+    or when the program is executed in an oversubscribed manner.
+
+    The following table summarizes the various behaviors of this method (where
+    {hostname} is the string returned by `here.hostname`, and {id} is the locale's
+    unique integer ID):
+
+  .. list-table::
+    :widths: 25 22 22
+    :header-rows: 1
+
+    * - Configuration
+      - locale ( `here.name` )
+      - sub-locale ( `here.gpus[0].name` )
+    * - normal execution
+      - {hostname}
+      - {hostname}-GPU0
+    * - oversubscribed execution
+      - {hostname}
+      - {hostname}-GPU0
+    * - oversubscribed execution (gasnet*)
+      - {hostname}-{id}
+      - {hostname}-{id}-GPU0
```

Comment:
This is a nice summary! I would prefer the above information to be formatted in a more human-friendly way so that the user reading the source code can understand it easily. Perhaps three bullet points like:
* oversubscribed with gasnet(*): {hostname}-{id} and {hostname}-{id}-GPU0

Reply:
I'm sort of inclined to keep this formatted as a table (rather than a bulleted list) because it makes it is more clear that each column is describing something different. I.e. the first column has the locale behavior, and the second has the sub-locale behavior. I don't think the separation is very easy to see with a bulleted list. 

The unformatted source doesn't look as good with the table, but I'm not sure how many users look at it directly. Let me know what you think...